### DISCUSSION TITLE
Rivet 2 alpha releases

### DISCUSSION THREAD

**Discussion by scottanthonymurray** [indiana-university/rivet-source#467] [Feature and enhancement requests]
Use this thread to discuss Rivet 2 alpha releases.

https://v2.rivet.iu.edu/docs/components/

**Comment by basham** [comment]
The [new Shadow classes](https://v2.rivet.iu.edu/docs/utilities/shadow/) look helpful. However, consider adding two layers of box-shadow, rather than one. The recent article [*Design detail: crafting better shadows for interaction*](https://uxdesign.cc/design-detail-crafting-better-shadows-for-interaction-b94796a29664) does an excellent job of explaining why and how to do this.

**Reply by levimcg** [reply]
Oooh, this is nice. I like the double shadow set up. It does looks so much more natural. Definitely happy to keep this conversation going. It would be a really nice, low-effort improvement 👍🏻

**Reply by pulsix** [reply]
Great post! I completely agree with the author. See also related articles here https://novini247.com/

**Comment by jameswilson** [comment]
From the initial [blog post](https://rivet.iu.edu/blog/rivet-2/#rivet-2-0) announcing Rivet 2:

> By the release of 2.0, Rivet will have feature parity with the Web Style Guide.

Is there any plan to put together a page in Rivet 2 docs that outlines a migration map from Web Style Guide Chunks to Rivet components?  This would be extremely helpful to have.

The majority of the [chunk types in the Web style guide pattern library](https://styleguide.iu.edu/pattern-library/chunks/) have replacements in Rivet 2 (bravo!). However,  there are still a few that I've found are missing or do not have a clear migration path to [Rivet 2 components list](https://v2.rivet.iu.edu/docs/components/).

Here is a list of my findings for the IU Libraries site:

* Rivet has no carousel component.  The **Feature Carousel** chunk is used extensively on IU Libraries website for longer lists of cards, such as for [Spaces](https://libraries.indiana.edu/spaces), as well as other Features listed on several Division landing pages.  I cannot find this chunk type documented on the Web Style Guide website, however the code to support it still exists in the 3.2.x Framework CSS and JS. It appears that the “Card” component (which replaces Feature chunk) is to be used in a grid layout, but I haven't seen anything indicating whether these could be placed into a carousel.

* Similar to the Feature Carousel, there is no replacement for the **News Ticker** component. Again this seems to be an undocumented (deprecated?) feature that is still supported in code. This is used on IU Libraries and Lilly Library homepage above the fold, as well as on the https://www.indiana.edu/ homepage.

* [**Panel**](https://styleguide.iu.edu/pattern-library/chunks/panels.html) chunks are not supported in Rivet. Need suggestions and recipes on how best to replace these using Rivet. The basic Panel might be fairly easy to replicate in Rivet, however the **Interactive Panel** much less so.  Would it be a Button, or something else?

* [**Image Essay**](https://styleguide.iu.edu/pattern-library/chunks/images/image-essays.html) chunk type doesn't have a similar component in Rivet.  Need suggestions on what will replace these. Cards? Note: IU Libraries website doesn't utilize this Chunk type. It is just something I noticed in my analysis mappings from Chunks to Components.

* [**Image Slideshow**](https://styleguide.iu.edu/pattern-library/chunks/images/image-slideshow.html) chunk type -- similar to Feature Carousel, there isn't any Rivet-blessed slideshow/carousel script.

* [**Social Media Embeds**](https://styleguide.iu.edu/pattern-library/chunks/social-media.html) not really mentioned in Rivet as far as I can tell.  Would there be a global JS file that handles formatting these chunks without adding extra scripts?

* Suggested approach for `<figure>` + `<figcaption>` tags such as those used for [**Video / Audio**](https://styleguide.iu.edu/pattern-library/chunks/video-audio.html) chunk types.  I haven't tested this on a pure Rivet site, but could we expect the default styles to "just work" or would it be more desirable to have a clear mapping to `rvt-` classes & dom structure?

* Suggested approach for [**News & Events feeds**](https://styleguide.iu.edu/pattern-library/sections/news-events.html) with appropriate Rivet markup classes, grid layout, optional (collapsible image), etc. Maybe a new example page in the [Rivet Layouts](https://v2.rivet.iu.edu/docs/layouts/) section could work for this. Note: I do see a replacement component for the [calendar tile](https://v2.rivet.iu.edu/docs/components/calendar-tile/) used on event feeds, which is great!

* Replacements for all [Banner types](https://styleguide.iu.edu/pattern-library/sections/banners.html) including the text, image, video, text overlay, or text and image types.  Note: I didn't spend a lot of time here trying to come up with a 1-to-1 mapping of each banner type, I know there are several new components such as the Hero and the Billboard components, but I'm not sure that all these elements are covered.

* I was pleased to see that Rivet _does have a recipe_ for **Footer + Resource links + Social Links**, which may have to replace the [**Belt**](https://styleguide.iu.edu/pattern-library/sections/belt.html), footer, and other elements at the bottom of the page. Rivet also provides a grid system which could be leveraged to put together a custom belt section replacement. This seems to leave us open to the possibility to improve upon the colophon area of the page, or remove it and standardize with the Rivet footer.

Note, this list is not exhaustive, and the analysis above is a bit of work in progress, that I thought I'd share in case someone else is also doing a similar effort.

**Reply by levimcg** [reply]
Hey, @jameswilson Thanks for this feedback!

Since the web style guide is primary related to the web framework, some elements of the style guide may not be represented 1-for-1 in rivet 2, but any future major version of the web framework would likely include its own rivet-based chunks that are specific to marketing websites. 

That said, Rivet is meant to be a platform for designing and building interfaces, so it will change and grow over time as we get feedback from the community. This list is really helpful to our team, so thank you :raised_hands:

> Is there any plan to put together a page in Rivet 2 docs that outlines a migration map from Web Style Guide Chunks to Rivet components? This would be extremely helpful to have.

This is a really great idea. While we don't have concrete plans for a migration guide/map right now, I think this is definitely something we can look at doing in the future. You're list here gives us a really great start ✅
